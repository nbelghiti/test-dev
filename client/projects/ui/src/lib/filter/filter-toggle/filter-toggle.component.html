<div class="c-filter-toggle" [ngClass]="{
	'c-filter-toggle--inline': inline,
	'c-filter-toggle--borderless': borderless,
	'c-filter-toggle--compact': compact,
	'c-filter-toggle--transparent': background === 'transparent',
	'c-filter-toggle--grey': background === 'grey',
	'c-filter-toggle--large': large
}">
	<span *ngIf="label" class="c-filter-toggle__label">{{label}}</span>

	<button class="c-filter-toggle__trigger" type="button">
		<span class="c-filter-toggle__current-option">{{currentOption}}</span>

		<div class="c-filter-toggle__icon">
			<ui-icon name="caret-down" size="small"></ui-icon>
		</div>
	</button>
</div>
