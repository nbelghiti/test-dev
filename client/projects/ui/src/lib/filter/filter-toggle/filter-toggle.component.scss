// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: User Info
// =============================================================================
.c-filter-toggle {
	display: flex;
	flex-direction: column;

	&:hover {

		.c-filter-toggle__trigger {
			border: $border-width-base solid $border-color-black;
		}
	}
}

// =============================================================================
// :: Elements
// =============================================================================
.c-filter-toggle__label {
	margin-bottom: 0.2rem;

	font-size: $font-size-2x-small;
	line-height: $line-height-base;

	color: $text-color-grey-46;
}

.c-filter-toggle__trigger {
	display: flex;
	justify-content: space-between;
	align-items: center;
	padding: 0.6rem 2rem;

	white-space: nowrap;

	background: none;
	border: $border-width-base solid $border-color-base;
	background-color: $background-color-base;

	transition:
		border $animation-base,
		background-color $animation-base,
		box-shadow $animation-base;

	cursor: pointer;

	&:focus {
		border: $border-width-base solid $border-color-black;
	}

	&:hover {

		.c-filter-toggle__current-option {
			text-decoration: underline;
		}
	}
}

.c-filter-toggle__current-option {
	overflow: hidden;
	margin-right: 1rem;

	font-weight: $font-weight-ultrabold;
	font-size: $font-size-2x-small;
	text-overflow: ellipsis;
	white-space: nowrap;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-filter-toggle--inline {
	flex-direction: row;
	align-items: center;

	.c-filter-toggle__label {
		margin-right: 1rem;
		margin-bottom: 0;

		color: $text-color-base;
	}

	.c-filter-toggle__trigger {
		flex-grow: 1;
	}
}

.c-filter-toggle--borderless {

	.c-filter-toggle__trigger {
		border-color: transparent;
	}

	&:hover {

		.c-filter-toggle__trigger {
			border-color: transparent;
			background-color: $background-color-grey-96;
		}
	}
}

.c-filter-toggle--compact {

	.c-filter-toggle__trigger {
		padding: 0 1rem;
	}
}

.c-filter-toggle--transparent {

	.c-filter-toggle__trigger {
		background-color: transparent;
	}
}

.c-filter-toggle--grey {

	.c-filter-toggle__trigger {
		color: $text-color-grey-46;
		background-color: $background-color-grey-96;

		&:focus {
			color: $text-color-base;
		}
	}

	&.c-filter-toggle--borderless:hover {

		.c-filter-toggle__trigger {
			box-shadow: 0 0 0 $border-width-base $border-color-black;
		}
	}
}

.c-filter-toggle--large {

	.c-filter-toggle__current-option {
		font-size: $font-size-1x-large;

		&:focus {
			color: $text-color-base;
		}
	}

	.c-filter-toggle__trigger {
		padding: 0;

		border: none;

		&:hover {
			border: none;
		}
	}
}
