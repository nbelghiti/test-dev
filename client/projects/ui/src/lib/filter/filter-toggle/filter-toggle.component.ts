import { Component, Input } from '@angular/core';

import { FilterToggleBackground } from './filter-toggle.component.types';

@Component({
	selector: 'ui-filter-toggle',
	templateUrl: './filter-toggle.component.html',
	styleUrls: ['./filter-toggle.component.scss']
})

export class FilterToggleComponent {
	@Input() label: string;
	@Input() currentOption: string = '- All';
	@Input() inline?: boolean = false;
	@Input() borderless?: boolean = false;
	@Input() compact?: boolean = false;
	@Input() large?: boolean = false;
	@Input() background?: FilterToggleBackground = FilterToggleBackground.default;
}
