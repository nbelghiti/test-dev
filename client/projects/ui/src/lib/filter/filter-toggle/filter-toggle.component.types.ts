export enum FilterToggleBackground {
	default = '',
	transparent = 'transparent',
	grey = 'grey',
}
