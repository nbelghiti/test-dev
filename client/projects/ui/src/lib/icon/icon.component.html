<i class="c-icon__glyph {{ size && 'c-icon__glyph--' + size }} {{ type && 'c-icon__glyph--' + type }} fas fa-{{ name }}"></i>
