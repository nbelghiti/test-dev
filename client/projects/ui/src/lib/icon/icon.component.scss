// =============================================================================
// :: Imports
// =============================================================================
@import "global";

// =============================================================================
// :: Icon
// =============================================================================
:host {
	display: inline-flex;
	justify-content: center;
	align-items: center;

	font-size: inherit;
}

// =============================================================================
// :: Elements
// =============================================================================
.c-icon__glyph {
	line-height: inherit;

	cursor: pointer;
}

// =============================================================================
// :: Modifiers
// =============================================================================
.c-icon__glyph--large {
	font-size: 2rem;
}

.c-icon__glyph--small {
	font-size: 1.4rem;
}

.c-icon__glyph--tiny {
	font-size: 1.2rem;
}

.c-icon__glyph--success {
	color: $ui-color-success-dark;
}

.c-icon__glyph--error {
	color: $ui-color-error-dark;
}

.c-icon__glyph--warning {
	color: $ui-color-warning-dark;
}

.c-icon__glyph--info {
	color: $ui-color-info-dark;
}
