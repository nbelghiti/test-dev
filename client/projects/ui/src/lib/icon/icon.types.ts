export enum IconName {
	archive = 'archive',
	arrowDown = 'arrow-down',
	arrowLeft = 'arrow-left',
	arrowRight = 'arrow-right',
	arrowRightAlt = 'arrow-alt-circle-right',
	arrows = 'arrows-alt-h',
	arrowUp = 'arrow-up',
	bell = 'bell',
	bellSlash = 'bell-slash',
	book = 'book',
	brackets = 'brackets',
	calendar = 'calendar',
	caretDown = 'caret-down',
	caretLeft = 'caret-left',
	caretRight = 'caret-right',
	caretUp = 'caret-up',
	check = 'check',
	checkCircle = 'check-circle',
	chevronDown = 'chevron-down',
	chevronLeft = 'chevron-left',
	chevronRight = 'chevron-right',
	chevronUp = 'chevron-up',
	clock = 'clock',
	clone = 'clone',
	close = 'times-circle',
	code = 'code',
	codeBranch = 'code-branch',
	cube = 'cube',
	download = 'download',
	edit = 'edit',
	excel = 'file-excel',
	exclamation = 'exclamation',
	exclamationCircle = 'exclamation-circle',
	expand = 'expand',
	externalLink = 'external-link-alt',
	file = 'file-alt',
	home = 'home',
	horizontalEllipsis = 'ellipsis-h',
	industry = 'industry',
	infoCircle = 'info-circle',
	invisible = 'eye-slash',
	marker = 'map-marker-alt',
	minus = 'minus',
	pdf = 'file-pdf',
	pen = 'pen',
	plus = 'plus',
	print = 'print',
	search = 'search',
	settings = 'cog',
	signOut = 'sign-out-alt',
	snowflake = 'snowflake',
	spinner = 'spinner',
	star = 'star',
	sync = 'sync-alt',
	table = 'table',
	times = 'times',
	trash = 'trash',
	upload = 'upload',
	user = 'user',
	users = 'users',
	visible = 'eye',
	plusSquare = 'plus-square',
	minusSquare = 'minus-square',
}

export enum IconType {
	success = 'success',
	error = 'error',
	warning = 'warning',
	info = 'info',
}

export enum IconSize {
	large = 'large',
	small = 'small',
	tiny = 'tiny',
}
