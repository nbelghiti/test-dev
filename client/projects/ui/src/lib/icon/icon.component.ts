import { Component, HostBinding, Input } from '@angular/core';

import { IconName, IconSize, IconType } from './icon.types';

@Component({
	selector: 'ui-icon',
	templateUrl: './icon.component.html',
	styleUrls: ['./icon.component.scss']
})

export class IconComponent {
	@Input() public name: IconName;
	@Input() public type: IconType = null;
	@Input() public size: IconSize = null;

	@HostBinding('class.c-icon') public className = true;
}
