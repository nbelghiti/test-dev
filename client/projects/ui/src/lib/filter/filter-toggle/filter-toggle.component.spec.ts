import { ComponentFixture, TestBed, async } from '@angular/core/testing';

import { IconComponent } from '../../icon/icon.component';

import { FilterToggleComponent } from './filter-toggle.component';

describe('UserInfoComponent', () => {
	let component: FilterToggleComponent;
	let fixture: ComponentFixture<FilterToggleComponent>;

	beforeEach(async(() => {
		TestBed.configureTestingModule({
			declarations: [
				FilterToggleComponent,
				IconComponent,
			],
		})
		.compileComponents();
	}));

	beforeEach(() => {
		fixture = TestBed.createComponent(FilterToggleComponent);
		component = fixture.componentInstance;
		fixture.detectChanges();
	});

	it('should create', () => {
		expect(component).toBeTruthy();
	});

	it('Should render correctly', () => {
		expect(fixture).toMatchSnapshot();
	});
});

