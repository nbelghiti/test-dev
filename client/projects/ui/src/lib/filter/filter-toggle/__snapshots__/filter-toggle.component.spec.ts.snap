// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoComponent Should render correctly 1`] = `
<ui-filter-toggle
  background=""
  borderless="false"
  compact="false"
  currentOption={[Function String]}
  inline="false"
  large="false"
>
  <div
    class="c-filter-toggle"
    ng-reflect-klass="c-filter-toggle"
    ng-reflect-ng-class="[object Object]"
  >
    
    <button
      class="c-filter-toggle__trigger"
      type="button"
    >
      <span
        class="c-filter-toggle__current-option"
      >
        - All
      </span>
      <div
        class="c-filter-toggle__icon"
      >
        <ui-icon
          class="c-icon"
          name="caret-down"
          ng-reflect-name="caret-down"
          ng-reflect-size="small"
          size="small"
        >
          <i
            class="c-icon__glyph c-icon__glyph--small  fas fa-caret-down"
          />
        </ui-icon>
      </div>
    </button>
  </div>
</ui-filter-toggle>
`;
